src/common/src/array/list_array.rs

```diff
@@ -431,118 +376,189 @@ impl ListValue {
     ) -> memcomparable::Result<Self> {
         let bytes = serde_bytes::ByteBuf::deserialize(deserializer)?;
         let mut inner_deserializer = memcomparable::Deserializer::new(bytes.as_slice());
-        let mut values = Vec::new();
+        let mut builder = datatype.create_array_builder(0);
         while inner_deserializer.has_remaining() {
-            values.push(memcmp_encoding::deserialize_datum_in_composite(
+            builder.append(memcmp_encoding::deserialize_datum_in_composite(
                 datatype,
                 &mut inner_deserializer,
             )?)
         }
-        Ok(Self::new(values))
+        Ok(Self::new(builder.finish()))
+    }
+
+    // Used to display ListValue in explain for better readibilty.
+    pub fn display_for_explain(&self) -> String {
+        // Example of ListValue display: ARRAY[1, 2, null]
+        format!(
```

Comment:
Is this okay?

```suggestion
    pub fn display_for_explain(&self) -> impl Display + '_ {
        // Example of ListValue display: ARRAY[1, 2, null]
        format_args!(
```

Reply:
I have tried but was running into lifetime issue. 🥹